marshal/consumer.go

```diff
@@ -145,7 +145,15 @@ func (m *Marshaler) NewConsumer(topicNames []string, options ConsumerOptions) (*
 					}
 				}
 			}
+
+			// this check needs to be after iterating all partitions in a topic
+			if len(c.claimedTopics) >= options.MaximumClaims {
```

Comment:
Doesn't this also have to check whether we're in topic claim mode or not? Otherwise, using `MaximumClaims` is not valid, right?